### PR TITLE
Add a congestionWindowInBytes method to Envoy::Network::Connection

### PR DESCRIPTION
**Pull request by wu-bin**
Add a congestionWindowInBytes() method to Envoy::Network::Connection.
- QUIC connections get the congestion window from QuicSentPacketManager::GetCongestionWindowInBytes().
- TCP connections get the congestion window from tcp_info.tcpi_snd_cwnd.

Also Implement Envoy::Network::Connection::lastRoundTripTime() for QUIC connections.

Commit Message: Add a congestionWindowInBytes method to Envoy::Network::Connection
Risk Level: Low
Testing: Unit test
Platform Specific Features: Getting congestion window for TCP connections may not be supported by all platforms.

### CODE REVIEW

File: envoy/api/os_sys_calls.h [hunk 1]
```diff
@@ -19,6 +19,9 @@ namespace Api {
 
 struct EnvoyTcpInfo {
   std::chrono::microseconds tcpi_rtt;
+  // Congestion window, in bytes. Note that posix's TCP_INFO socket option returns cwnd in packets,
+  // we multiply it by 1460(the default TCP MSS) to get bytes.
```
Comment (danzh2010) [inline envoy/api/os_sys_calls.h:23]: 1460 default MSS seems fine for internet traffic. I'm wondering if it worth to make it configurable for user controlled network or ENVOY_BUG if TCP MSS is not 1460. @mattklein123 for thoughts?

File: source/common/network/connection_impl.cc [hunk 2]
```diff
@@ -788,6 +788,10 @@ void ConnectionImpl::configureInitialCongestionWindow(uint64_t bandwidth_bits_pe
   return transport_socket_->configureInitialCongestionWindow(bandwidth_bits_per_sec, rtt);
 }
 
+absl::optional<uint64_t> ConnectionImpl::congestionWindowInBytes() const {
+  return socket_->congestionWindowInBytes();
```
[the remark this answers is not in the document]
Comment (mattklein123) [reply source/common/network/connection_impl.cc:792]: No real opinion. Whatever you both think is fine with me though I agree about being consistent.

[the remark this answers is not in the document]
Comment (wu-bin) [reply source/common/network/connection_impl.cc:792]: Thanks Matt. I talked with Dan offline and we'll leave it as is.

**Review by danzh2010 — APPROVED** [review]
LGTM

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**